### PR TITLE
Charging DC purchase percent by host

### PR DESCRIPTION
**Pull request by ambuznego**
With `HOST_DC_CHARGE_PERCENT` as `.2` 
Screen Shot 2021-09-13 at 6 07 29 PM

### CODE REVIEW

File: assets/js/components/billing/PurchaseCreditModal.jsx [hunk 1]
```diff
@@ -307,183 +367,261 @@ class PurchaseCreditModal extends Component {
...
-            )
-          }
+          {this.state.countUSD > 0 && (
+            <div style={styles.costContainer}>
+              <Text style={{ color: "#4091F7", marginTop: -5 }}>Cost:</Text>
+              <Text style={styles.costNumber}>
+                USD {countUSD && parseFloat(countUSD).toFixed(2)}
+              </Text>
+            </div>
+          )}
```
Comment (ambuznego) [inline assets/js/components/billing/PurchaseCreditModal.jsx:384]: main change

File: assets/js/components/billing/PurchaseCreditModal.jsx [hunk 2]
```diff
@@ -69,235 +77,287 @@ class PurchaseCreditModal extends Component {
...
       this.setState({
         countDC: e.target.value,
-        countUSD: e.target.value / 100000,
+        countUSD:
+          e.target.value * 0.00001 +
+          e.target.value *
+            0.00001 *
+            (window.host_dc_charge_percent ||
+              process.env.HOST_DC_CHARGE_PERCENT ||
+              0),
```
Comment (ambuznego) [inline assets/js/components/billing/PurchaseCreditModal.jsx:152]: main change

File: assets/js/components/billing/PurchaseCreditModal.jsx [hunk 3]
```diff
@@ -69,235 +77,287 @@ class PurchaseCreditModal extends Component {
...
+          alignItems: "center",
+        }}
+      >
+        <Text>{`1 DC = 24 Byte Packet = $${
+          window.host_dc_charge_percent || process.env.HOST_DC_CHARGE_PERCENT
+            ? 0.00001 +
+              (window.host_dc_charge_percent ||
+                process.env.HOST_DC_CHARGE_PERCENT) *
+                0.00001
+            : 0.00001
```
Comment (ambuznego) [inline assets/js/components/billing/PurchaseCreditModal.jsx:356]: main change #2

File: assets/webpack.config.js [hunk 4]
```diff
@@ -111,6 +111,7 @@ module.exports = function (env) {
         "RELEASE_BLOG_LINK",
         "MAPBOX_PRIVATE_KEY",
         "MAPBOX_STYLE_URL",
+        "HOST_DC_CHARGE_PERCENT",
```
Comment (vicmgs) [inline assets/webpack.config.js:114]: dont think we need this since this webpack config is only used for non self-hosted

Comment (ambuznego) [reply assets/webpack.config.js:114]: ah! good point.

File: assets/js/components/billing/PurchaseCreditModal.jsx (hunk 2 above)
Comment (vicmgs) [reply assets/js/components/billing/PurchaseCreditModal.jsx:152]: we probably just need: 

window.host_dc_charge_percent || 0

File: assets/js/components/billing/PurchaseCreditModal.jsx (hunk 3 above)
Comment (vicmgs) [reply assets/js/components/billing/PurchaseCreditModal.jsx:356]: same here

**Review by vicmgs — APPROVED** [review]

**Pull request merged**